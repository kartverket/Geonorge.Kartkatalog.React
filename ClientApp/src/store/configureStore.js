import { createBrowserHistory } from 'history';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { routerMiddleware } from 'connected-react-router';
import rootReducer from '../reducers'

//export const history = createBrowserHistory()

//export default function configureStore(initialStore) {

// In development, use the browser's Redux dev tools extension if installed
// if (isDevelopment && typeof window !== 'undefined' && window.devToolsExtension) {
//   enhancers.push(window.devToolsExtension());
//  enhancers.push(window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ? window.__REDUX_DEVTOOLS_EXTENSION__ : compose);
// }
/*if (isDevelopment && typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
  enhancers.push(window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__());
}*/


//routerMiddleware(history), // for dispatching history actions

const initialState = {};

const middleware = [thunk];

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  rootReducer,  // root reducer with router state
  initialState,
  compose(
    applyMiddleware(...middleware),
    thunk,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )
)

export default store;

//const history = createBrowserHistory();

//export history;

