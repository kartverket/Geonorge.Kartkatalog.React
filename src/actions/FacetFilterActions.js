import { ADD_SELECTEDFACET, REMOVE_SELECTEDFACET } from '../actions/types';

export const addSelectedFacet = (facet, facetField) => dispatch => {
	dispatch({
		type: ADD_SELECTEDFACET,
		payload: facet,
		facetField: facetField
	})
}

export const removeSelectedFacet = (facet, facetField) => dispatch => {
	dispatch({
		type: REMOVE_SELECTEDFACET,
		payload: facet,
		facetField: facetField
	})
}
