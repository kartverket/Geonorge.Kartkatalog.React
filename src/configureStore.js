import { createBrowserHistory } from 'history'
import { routerMiddleware } from 'connected-react-router'
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import createRootReducer  from './reducers';

export const history = createBrowserHistory()



export default function configureStore(preloadedState) {
	const middleware = [thunk];
	const history = {};
	const store = createStore(
		createRootReducer(history), 
		preloadedState,
		compose (
			applyMiddleware(...middleware, routerMiddleware(history)),
			window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
			) 
		);

	return store;
}

