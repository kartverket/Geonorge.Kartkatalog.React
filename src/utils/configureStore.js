import { createBrowserHistory } from 'history'
import { routerMiddleware } from 'connected-react-router'
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import createRootReducer from '../reducers';

export const history = createBrowserHistory()

const composeEnhancers = composeWithDevTools({
	// options like actionSanitizer, stateSanitizer
});

export default function configureStore(preloadedState) {
	const middleware = [thunk];
	const history = {};
	const store = createStore(
		createRootReducer(history),
		preloadedState,
		composeEnhancers(
			applyMiddleware(...middleware, routerMiddleware(history)),
		)
	);

	return store;
}

