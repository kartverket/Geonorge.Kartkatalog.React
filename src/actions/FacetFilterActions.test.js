import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

// Actions to be tested
import * as facetFilterActions from './FacetFilterActions';


const middlewares = [thunk] // add your middlewares like `redux-thunk`
const mockStore = configureStore(middlewares);
const store = mockStore();

describe('facet_filter_actions', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
    });

    describe('updateSelectedFacets', () => {
        test('Dispatches the correct action and payload', () => {
            const facets = [
                {
                    facets: [
                        { Name: 'dataset' },
                        { Name: 'service' }
                    ],
                    facetField: 'type'
                },
                {
                    facets: [
                        { Name: 'Befolkning' },
                        { Name: 'Energi' }
                    ],
                    facetField: 'theme'
                }
            ]
            store.dispatch(facetFilterActions.updateSelectedFacets(facets));
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    describe('updateSelectedFacetsForFacetField', () => {
        test('Dispatches the correct action and payload', () => {
            const facets = [
                { Name: 'dataset' },
                { Name: 'service' }
            ];
            const facetField = 'type';
            store.dispatch(facetFilterActions.updateSelectedFacetsForFacetField(facets, facetField));
            expect(store.getActions()).toMatchSnapshot();
        });
    });
});
