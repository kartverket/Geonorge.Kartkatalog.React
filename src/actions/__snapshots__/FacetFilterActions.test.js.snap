// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`facet_filter_actions addSelectedFacet Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "facetField": "type",
    "payload": Object {
      "Count": 287,
      "Name": "dataset",
      "NameTranslated": "Datasett",
    },
    "type": "ADD_SELECTEDFACET",
  },
]
`;

exports[`facet_filter_actions removeSelectedFacet Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "facetField": "type",
    "payload": Object {
      "Count": 287,
      "Name": "dataset",
      "NameTranslated": "Datasett",
    },
    "type": "REMOVE_SELECTEDFACET",
  },
]
`;

exports[`facet_filter_actions updateSelectedFacets Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "facetField": "type",
        "facets": Array [
          Object {
            "Name": "dataset",
          },
          Object {
            "Name": "service",
          },
        ],
      },
      Object {
        "facetField": "theme",
        "facets": Array [
          Object {
            "Name": "Befolkning",
          },
          Object {
            "Name": "Energi",
          },
        ],
      },
    ],
    "type": "UPDATE_SELECTEDFACETS",
  },
]
`;

exports[`facet_filter_actions updateSelectedFacetsForFacetField Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "facetField": "type",
    "payload": Array [
      Object {
        "Name": "dataset",
      },
      Object {
        "Name": "service",
      },
    ],
    "type": "UPDATE_SELECTED_FACETS_FOR_FACET_FIELD",
  },
]
`;
